### DISCUSSION TITLE
config.init_weight_method = 'decay' vs original

### DISCUSSION THREAD

**Discussion by gateway** [rkhamilton/vqgan-clip-generator#29] [Q&A] [answered]
I did a test this morning with the two options and same seed, iterations and save frames.. I'm not seeing a lot of huge differences.. 

![image](https://user-images.githubusercontent.com/519563/138503680-81d98fa9-9a2b-43c8-a471-e5f580594ceb.png)

**Comment by rkhamilton** [comment, marked as answer]
What config.init_weight are you using for these? The two methods respond very differently as init_weight increases, particularly with restyle_video.
With the 'decay' method, the more iterations you run, the less it cares what was in the init_image.
With the 'original' method, the important of the initial image is the same.

With 'decay' at higher weights (>5), and lower iterations (<50) tends to produce a flat gray/white image.
With 'original' at higher weights (>1), and lower iterations, it tends to leave you with the original image.
Here is 50 iterations, init_weight 5.0.
![image](https://user-images.githubusercontent.com/8406344/138508952-8f777f51-8ab0-44d4-8a9b-b7b748bb68fa.png)

I do most of my evaluations with restyle video, and differences are more noticable there, where they can build up over frames. In general, if you want your output image to be closer to the input image, 'original' works better. I tend to want my output videos to look closer to the inputs, so I run a lot at lower iterations, medium weights.

**Reply by gateway** [reply]
yea so I didn't really change weight or iterations but good to know..  I ran some quick tests here are some results..

![image](https://user-images.githubusercontent.com/519563/138512012-18cb0d15-6c85-4568-ad9f-21c570cc53cc.png)

I think now with the meta data and then writing a script to go though each value incremental will be great especially to create a montage or a static html page with all the info.. for me as an artist first being able to visualize variations are a key and then I usually choose that image with its settings to try other things.

**Reply by gateway** [reply]
oh is their a upper bounds area for weight and iterations when things really dont change that much anymore in your opinion ? also does save image every really do anything outside of being able to visualize the image changing while your looking at it?

**Reply by rkhamilton** [reply]
When creating a single image, the only purpose of save_every is to periodically dump an image to disk so you can look at what’s happening, rather than waiting to the end. I often set it to none because I don’t care.

As far as upper bounds, for a single image it seems like 150-200 is about where things totally stop. In fact, if you use config.adaptiveLR = True, then the learning rate will slow and you will not get jitter anymore in the final frames.
As far as upper bounds for weights, I don’t think so. The prompt weights are all relative to each other, so (horse:1.0 | dog:1.0, image_prompt.jpg:1.0) should do the same as (horse:0.1 | dog:0.1, image_prompt:0,1). Weights on the init_image are also unbounded, and also sort of work relative to the others. What those weights do is multiply the error from each prompt, or from the init_image loss function. When you see those scrolling numbers as images are being saved, the last set of numbers says something like “loss for each prompt.” Those final numbers are the loss from each prompt, and the init_image. The training process is trying to drive the sum of those numbers down. Increasing a weight makes the loss bigger, train() will try harder to drive that loss down.

**Reply by rkhamilton** [reply]
Speaking of exploring parameters, you might find the two attached scripts useful. They use combinations of parameters to generate images. single_images_sweep.py creates images for combinations of three parameters.
restyle_video_sweep is more complicated. I basically pretends your init_image is a bunch of video stills (imagine a video where nothing changes on screen), then runs the restyle_video_frames algorithm for a certain number of video frames, then saves the result. The idea was I wanted to know what the video would look like a second or so into it, after things settle down. This script runs that process for all of the combinations of parameters. You can't run them all at once because it would create tens of thousands of images, but you can pick the ones you want to explore. Hope this is useful.

Cheers
[scripts.zip](https://github.com/rkhamilton/vqgan-clip-generator/files/7401364/scripts.zip)
.
